controllers/cache.go

```diff
@@ -104,33 +105,54 @@ type eventToChannelHandler struct {
 }
 
 func (e *eventToChannelHandler) OnAdd(obj interface{}) {
-	e.queueEventForObject(obj)
+	e.queueEventForObject(nil, obj)
 }
 
 func (e *eventToChannelHandler) OnUpdate(oldobj, obj interface{}) {
-	e.queueEventForObject(obj)
+	e.queueEventForObject(oldobj, obj)
 }
 
 func (e *eventToChannelHandler) OnDelete(obj interface{}) {
-	e.queueEventForObject(obj)
+	e.queueEventForObject(nil, obj)
 }
 
 // queueEventForObject sends the event onto the channel
-func (e *eventToChannelHandler) queueEventForObject(o interface{}) {
-	if o == nil {
+func (e *eventToChannelHandler) queueEventForObject(oldObj, newObj interface{}) {
+	if newObj == nil {
 		// Can't do anything here
 		return
 	}
-	obj, ok := o.(client.Object)
+	new, ok := newObj.(client.Object)
 	if !ok {
 		return
 	}
-	if obj.GetName() != e.name {
+	if new.GetName() != e.name {
 		// Not the right object, skip
 	}
 
+	if oldObj != nil {
+		old, ok := oldObj.(client.Object)
+		if !ok {
+			return
+		}
+
+		// We'd like to compare both objects ignoring changes in managedFields and resourceVersion
+		// Nither of those fields should be applied, so it is ok to strip
+		// Mitigating issue: https://github.com/kubernetes/kubernetes/issues/100024
+
+		old.SetManagedFields(nil)
+		new.SetManagedFields(nil)
+		old.SetResourceVersion("")
+		new.SetResourceVersion("")
```

Comment:
These operations add a side-effect that wasn't previously there. We should copy old and new before modifying them.

Reply:
You are right. It is now updated to work with resource copy.